File: 2-expressive-puzzlers/puzzle-1/Oddity.scala
object Oddity_S {
  def main(args: Array[String]) {
    println(isOdd(2))
    println(isOdd(1))
    println(isOdd(0))
    println(isOdd(-1))
    println(isOdd(-2))
  }

  def isOdd(i: Int): Boolean = {
    i % 2 == 1
  }
}

File: 2-expressive-puzzlers/puzzle-1/Oddity.java
public class Oddity {
    public static boolean isOdd(int i) {
        System.out.println(i + ": " + (i % 2));
        return (i % 2) == 1;
    }

    public static void main(String[] args) {
        System.out.println(isOdd(2));
        System.out.println(isOdd(1));
        System.out.println(isOdd(0));
        System.out.println(isOdd(-1));
        System.out.println(isOdd(-2));
    }
}